packages/website/docs/docs/block-types.md

```diff
@@ -102,4 +102,244 @@ type DefaultBlockProps = {
+
+::: sandbox {template=react-ts}
+
+```typescript /App.tsx
+import {
+  BlockNoteEditor,
+  defaultBlockSchema,
+  defaultProps,
+} from "@blocknote/core";
+import {
+  BlockNoteView,
+  useBlockNote,
+  createReactBlockSpec,
+  InlineContent,
+  ReactSlashMenuItem,
+  defaultReactSlashMenuItems,
+} from "@blocknote/react";
+import "@blocknote/core/style.css";
+import { RiImage2Fill } from "react-icons/ri";
+
+export default function App() {
+  // Configuration for a custom image block.
+  const ImageBlockConfig = {
+    type: "image",
+    propSchema: {
+      ...defaultProps,
+      src: {
+        default: "https://via.placeholder.com/1000",
+      },
+    },
+    containsInlineContent: true,
+    render: ({ block }) => (
+      <div
+        style={{
+        display: "flex",
+          flexDirection: "column",
+      }}>
+        <img
+          style={{
+            width: "100%",
+          }}
+          src={block.props.src}
+          alt={"Image"}
+          contentEditable={false}
+        />
+        <InlineContent />
+      </div>
+    ),
+  };
+
+  // Converts the image block config to a BlockSpec for BlockNote to use.
+  const ImageBlock = createReactBlockSpec(ImageBlockConfig);
+
+  // Creates a slash menu item for inserting an image block.
+  const insertImage = new ReactSlashMenuItem<
+    DefaultBlockSchema & { image: typeof ImageBlock }
+  >(
+    "Insert Image",
+    (editor) => {
+      const src: string | null = prompt("Enter image URL");
+      editor.insertBlocks(
+        [
+          {
+            type: "image",
+            props: {
+              src: src || "https://via.placeholder.com/1000",
+            },
+          },
+        ],
+        editor.getTextCursorPosition().block,
+        "after"
+      );
+    },
+    ["image", "img", "picture", "media"],
+    "Media",
+    <RiImage2Fill />,
+    "Insert an image"
+  );
+
+  // Creates a new editor instance.
+  const editor: BlockNoteEditor<
+    DefaultBlockSchema & { image: typeof ImageBlock }
+  > | null = useBlockNote({
+    // Tells BlockNote which blocks to use.
+    blockSchema: {
+      // Adds all default blocks.
+      ...defaultBlockSchema,
+      // Adds the custom image block.
+      image: ImageBlock,
+    },
+    slashCommands: [...defaultReactSlashMenuItems, insertImage],
+  });
+
+  // Renders the editor instance using a React component.
+  return <BlockNoteView editor={editor} />;
+}
+```
+
+:::
+
+### Block Config Objects
+
+To define a custom block type, we use a `ReactBlockConfig` object, for which you can see the definition below:
+
+```typescript
+type PropSchema = Record<
+  string,
+  {
+    default: string;
+    values?: string[];
+  };
+>
+
+type ReactBlockConfig<
```

Comment:
I think it will be a lot easier if we keep the term `BlockConfig` internally, or only in the context of `createReactBlockSpec`. I think we should focus first on saying you need to create a new block spec using `createReactBlockSpec`, and that that function takes an argument of:

```typescript
type: Type;
  propSchema: PropSchema;
  containsInlineContent: boolean;
  render: (props: {
    block: Block,
    editor: BlockNoteEditor
  }) => JSX.Element;
```